src/main/java/carsharing/customer/CustomerDAOImpl.java

```diff
@@ -0,0 +1,214 @@
+    private static final String SELECT_NAME_AND_COMPANY_ID = "SELECT name, company_id FROM car WHERE id = ?";
+    private static final String SELECT_NAME_BY_COMPANY = "SELECT name FROM company WHERE id = ?";
+    private static final String UPDATE_CUSTOMER_WITH_NULL_RENTED_CAR = "UPDATE customer SET rented_car_id = NULL WHERE id = ?";
+    private static final String SELECT_ID_BY_CAR_NAME = "SELECT id FROM car WHERE name = ?";
+    private static final String UPDATE_CUSTOMER = "UPDATE customer SET rented_car_id = ? WHERE id = ?";
+    private static final String SELECT_ID_BY_COMPANY = "SELECT id FROM company WHERE name = ?";
+    private static final String SELECT_RENTED_CAR = "SELECT rented_car_id FROM customer WHERE rented_car_id IS NOT NULL";
+    private static final String SELECT_ID_NAME_BY_CAR = "SELECT id, name FROM car WHERE company_id = ? ORDER BY id";
+    private static final String INSERT_INTO_CUSTOMER_BY_NAME = "INSERT INTO customer (name) VALUES ?";
+
+    private final DBManager DBManager;
+    private Set<Integer> rentedCars;
+    private List<Car> cars;
+    private int company_id;
+
+    public CustomerDAOImpl(String name) {
+        super(name);
+        DBManager = new DBManager();
+    }
+
+    @Override
+    public List<Customer> getCustomers() {
+        List<Customer> result = new ArrayList<>();
+        try (Connection connection = DBManager.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(SELECT_NAME_BY_CUSTOMER)) {
+            connection.setAutoCommit(true);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                result.add(new Customer(resultSet.getString(1)));
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return result;
+    }
+
+    @Override
+    public Customer getInfoAboutRentedCar(int customerId) {
+        Customer customer = null;
+        try (Connection connection = DBManager.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(SELECT_NAME_AND_CAR_ID)) {
+            connection.setAutoCommit(true);
+            preparedStatement.setInt(1, customerId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                customer = new Customer(resultSet.getString(1));
+                int carId = resultSet.getInt(2);
+                if (carId != 0) {
+                    setCompanyId(customer, carId);
+                    setCompany(customerId, resultSet, customer);
+                }
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return customer;
+    }
+
+    private void setCompanyId(Customer customer, int carId) {
+        try (Connection connection = DBManager.getConnection();
```

Comment:
We already have opened connection when using this method

Reply:
fixed